### PR TITLE
[Dogfooding] Extend existing IDE code style enforcement rules to EditorFeatures layer

### PR DESCRIPTION
**Pull request by mavasani**
**Strongly recommended to review commit-by-commit**

Builds on top of prior PRs that aim at dogfooding code style enforcement in IDE layers. No new rules are enforced here, just existing rules are now respected in `EditorFeatures` layer.

### CODE REVIEW

File: src/EditorFeatures/Core/Implementation/TextBufferAssociatedViewService.cs [hunk 1]
```diff
@@ -123,7 +123,9 @@ public static bool AnyAssociatedViewHasFocus(ITextBuffer textBuffer)
         }
 
         [Conditional("DEBUG")]
+#pragma warning disable CA1822 // Mark members as static - accesses instance members within #if DEBUG
```
Comment (sharwell) [inline src/EditorFeatures/Core/Implementation/TextBufferAssociatedViewService.cs:126]: 😕 Can you not just mark `OnTextViewClose` static at the same time?

Comment (mavasani) [reply src/EditorFeatures/Core/Implementation/TextBufferAssociatedViewService.cs:126]: Ah, I did not get a CA1822 on that method and assumed it needs instance access. Seems the analyzer is bailing out on that method as it has event like signature. I can remove this pragma and make both methods static.

File: src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs [hunk 2]
```diff
@@ -205,11 +205,11 @@ public async Task<ImmutableArray<CodeFixCollection>> GetFixesAsync(Document docu
             if (result.Count > 0 && _fixerPriorityMap.TryGetValue(document.Project.Language, out var fixersForLanguage))
             {
                 // sort the result to the order defined by the fixers
-                ImmutableDictionary<CodeFixProvider, int> priorityMap = fixersForLanguage.Value;
+                var priorityMap = fixersForLanguage.Value;
                 result.Sort((d1, d2) => GetValue(d1).CompareTo(GetValue(d2)));
 
                 int GetValue(CodeFixCollection c)
-                    => priorityMap.TryGetValue((CodeFixProvider)c.Provider, out var value) ? value : int.MaxValue;
+                    => priorityMap!.TryGetValue((CodeFixProvider)c.Provider, out var value) ? value : int.MaxValue;
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs:212]: don't love this.

File: src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsAdornmentTaggerProvider.cs [hunk 3]
```diff
@@ -78,9 +78,9 @@ protected object CreateToolTipContent(Workspace workspace, DiagnosticData diagno
         }
 
         protected virtual SnapshotSpan AdjustSnapshotSpan(SnapshotSpan span, int minimumLength)
-            => AdjustSnapshotSpan(span, minimumLength, int.MaxValue);
+            => AbstractDiagnosticsAdornmentTaggerProvider<TTag>.AdjustSnapshotSpan(span, minimumLength, int.MaxValue);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsAdornmentTaggerProvider.cs:81]: seems unnecessary.

Comment (mavasani) [reply src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsAdornmentTaggerProvider.cs:81]: Hmm code fix added it, and I think it adds simplifier annotations. Let me see if there is underlying bug in simplifier.

File: src/EditorFeatures/Core/Shared/Extensions/ITextBufferEditExtensions.cs [hunk 4]
```diff
@@ -28,8 +22,6 @@ public static ITextSnapshot ApplyAndLogExceptions(this ITextBufferEdit edit)
             }
             catch (Exception e) when (ErrorReporting.FatalError.ReportWithoutCrash(e))
             {
-                s_lastException = e;
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Shared/Extensions/ITextBufferEditExtensions.cs:31]: this looks like it shoudl be kept right?  it was for crash dump investigations it looks like.

File: src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs (hunk 2 above)
Comment (mavasani) [reply src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs:212]: It seems unfortunate that nullable warning is generated here, when the declaration explicitly initializes it to a non-null value. I can revert back the declaration to specify explicit type name and suppress "use var" diagnostic on that instance.

File: src/EditorFeatures/Core/Shared/Extensions/ITextBufferEditExtensions.cs (hunk 4 above)
Comment (mavasani) [reply src/EditorFeatures/Core/Shared/Extensions/ITextBufferEditExtensions.cs:31]: Sure, I can restore.

File: src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsAdornmentTaggerProvider.cs (hunk 3 above)
Comment (mavasani) [reply src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsAdornmentTaggerProvider.cs:81]: Yep, seems like simplifier is not marking this as unnecessary, hence the behavior. I have manually removed it.

**Pull request merged**